### DISCUSSION TITLE
Add/Modify Block Values

### DISCUSSION THREAD

**Discussion by derks** [gaphor/gaphor#3498] [General]
Hello all, I am new to Gaphor and relatively new to SysML. I'm working through the SysML Distilled book (excellent), and trying to recreate diagrams in the book using Gaphor.

I am having an issue understanding Block values in Gaphor... specifically, how to add values to a block. I can create a ValueType fine, however on a Block when I toggle the "Values" compartment, I see no way to add/modify values to the Block's values compartment.

<img width="524" alt="Gaphor Block Values" src="https://github.com/user-attachments/assets/f0365509-773a-4b42-bfdc-8bc8da3c50e6">

Any help would be much appreciated, and please don't hesitate to tell me to RTFD on this... as I looked through here and Gaphor docs and didn't seem to find any clear answer or reference to this.

Thank you!

**Comment by amolenaar** [comment]
Hi @derks, Thanks for the question.

You can create a ValueType and add a composite association:

<img width="252" alt="image" src="https://github.com/user-attachments/assets/6b1b5392-bba1-4418-b8f4-cb9fa08391d9">


PS. It's a good suggestion to implement diagrams from a book. I'll order a copy myself :smile:

**Reply by derks** [reply]
@amolenaar Thank you for the quick reply! This is helpful, and gets me moving forward. I am sure I will have more questions as I work through this however I realize I have a lot to learn here.

**Comment by mikekidner** [comment]
@derks / @amolenaar I'd started to do something similar in the past. I've made it available at this [repo](https://github.com/mikekidner/gaphor_distilled)
It's only small so far, but happy for folks to add to it.
<img width="1179" alt="Screen Shot 2024-11-20 at 12 35 57 pm" src="https://github.com/user-attachments/assets/088f2e2d-a8c5-44f3-879a-c0d34a50c59c">
